### PR TITLE
Mqtt application test

### PR DESCRIPTION
**Pull request by YannickWeber**

### CODE REVIEW

File: hivemq-mqtt-application-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <groupId>org.example</groupId>
+    <artifactId>hivemq-howToTestMQTT</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
```
Comment (SgtSilvio) [inline hivemq-mqtt-application-test/pom.xml:16]: does this work, doesn't it need to be 1.8?

File: hivemq-mqtt-application-test/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.eclipse.paho</groupId>
+            <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
+            <version>1.2.4</version>
```
Comment (SgtSilvio) [inline hivemq-mqtt-application-test/pom.xml:27]: missing scope

File: hivemq-mqtt-application-test/src/main/java/com/smartaquarium/equipment/Co2.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.smartaquarium.equipment;
```
Comment (SgtSilvio) [inline hivemq-mqtt-application-test/src/main/java/com/smartaquarium/equipment/Co2.java:1]: should be `com.hivemq.examples.smartaquarium...`

File: hivemq-mqtt-application-test/src/test/resources/logback-test.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    <root>
+        <appender-ref ref="CONSOLE"/>
+    </root>
+
+    <root level="OFF">
+        <appender-ref ref="CONSOLE"/>
+    </root>
+
+    <logger name="com.hivemq.client" level="OFF"/>
```
Comment (SgtSilvio) [inline hivemq-mqtt-application-test/src/test/resources/logback-test.xml:34]: why is this needed?

File: hivemq-mqtt-application-test/src/test/java/com/smartaquarium/mqtt/SmartAquariumClientIT.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+        testClient.subscribeWith()
+                .topicFilter(SmartAquariumClient.TEMPERATURE_TOPIC)
+                .qos(MqttQos.EXACTLY_ONCE)
+                .send();
+
+        when(temperatureSensor.getCelsius()).thenReturn(13.0f);
+        smartAquariumClient.publishTemperature();
+
+        final Mqtt3Publish mqtt3Publish = testClient.publishes(MqttGlobalPublishFilter.ALL).receive();
```
Comment (SgtSilvio) [inline hivemq-mqtt-application-test/src/test/java/com/smartaquarium/mqtt/SmartAquariumClientIT.java:89]: race condition, you publish before consuming

File: hivemq-mqtt-application-test/src/test/java/com/smartaquarium/mqtt/SmartAquariumClientIT.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+        assertNotNull(mqtt3Publish.getPayloadAsBytes());
+        final String payload = new String(mqtt3Publish.getPayloadAsBytes(), StandardCharsets.UTF_8);
+        assertEquals("13,0°C", payload);
```
Comment (SgtSilvio) [inline hivemq-mqtt-application-test/src/test/java/com/smartaquarium/mqtt/SmartAquariumClientIT.java:93]: is the comma right, should it be a point?

File: hivemq-mqtt-application-test/pom.xml (hunk 1 above)
Comment (YannickWeber) [reply hivemq-mqtt-application-test/pom.xml:16]: Yes ist does. Its how intellij generated it.
